### ISSUE TITLE
TrailingSemicolon: Incorrect results for single line nested ListLiterals and MapLiterals

### ISSUE THREAD

**Issue by ThomasR** [brigade/scss-lint#690]
In this code, the TrailingSemicolon linter incorrectly reports missing semicolons:
```scss
    $foo: ((1, 2));
    $bar: ((1 2));
    $baz: ("a": ("b", "c"));
    $baz: ("a": ("b" "c"));
```

---

I'm pretty sure this is caused by [trailing_semicolon.rb:24](https://github.com/brigade/scss-lint/blob/8859b8a1fcc4d8ed810ec1f79bea75afc40b993d/lib/scss_lint/linter/trailing_semicolon.rb#L24), and removing the line (and the `&&`) appears to solve all problems.
```ruby
      if (node.expr.is_a?(Sass::Script::Tree::ListLiteral) ||
          node.expr.is_a?(Sass::Script::Tree::MapLiteral)) &&
         !node_on_single_line?(node) # <-- here
        return check_semicolon(node)
      end
```

Am I missing something? I really don't understand why that check has been introduced.

**Comment by sds** [comment]
Thanks for the fix, @ThomasR, and for taking the time to improve the set of tests for this linter.

Merged in 20523e4.
